mdaviz/utils.py

```diff
@@ -63,16 +71,43 @@ def byte2str(byte_literal):
 
 
 def get_det(mda_file_data):
-    """det_dict = { index: [fieldname, pv, desc, unit]}"""
+    """
+    Extracts scan positioners and detectors from an MDA file data object.
+
+    This function processes an mda.scanDim object to extract its scanPositioner and scanDetector instances.
+    It organizes these instances into a dictionary, with their indexes as keys in the order of P0, P1,... Px, D01, D02,... DX.
+    P0 is a default scanPositioner object representing the point index. If additional positioners exist, they follow P0 in sequence.
+    The first detector is labeled D01 and subsequent detectors follow in numerical order.
+
+    Parameters:
+    - mda_file_data: An instance of an mda.scanDim object, which contains the MDA file data to be processed.
+
+    Returns:
+    - A tuple containing:
+      - A dictionary (D) where keys are indexes, mapping to either scanPositioner or scanDetector objects.
+        The dictionary is structured as {0: P0, 1: P1, ..., np: D01, np+1: D02, ..., np+nd: DX}.
+      - The index (first_pos) of the first positioner in the returned dictionary. This is 1 if a positioner
+        other than the default index positioner exists, otherwise 0.
+      - The index (first_det) of the first detector in the returned dictionary, which directly follows the last positioner.
+
+    Note:
+    - P0 is created by default and corresponds to the point index, described as an 'Index' scanPositioner object with predefined properties.
+    - np is the total number of positioners, nd the number of detectors, and npts the number of data points actually acquired.
+    """
+    from mda import scanPositioner
+
     D = {}
-    p_list = [
-        mda_file_data.p[i] for i in range(0, mda_file_data.np)
-    ]  # mda_file_data.np = number of positioners
-    d_list = [
-        mda_file_data.d[i] for i in range(0, mda_file_data.nd)
-    ]  # mda_file_data.nd = number of detectors
-    first_pos = 1 if mda_file_data.np else 0
-    first_det = mda_file_data.np + 1
+    print(f"\n\n{mda_file_data=}\n\n")
+
+    p_list = mda_file_data.p  # list of scanDetector instances
+    d_list = mda_file_data.d  # list of scanPositioner instances
+    np = (
+        mda_file_data.np
+    )  # number of pos; for reference: nd = mda_file_data.nd (# of det)
+    npts = mda_file_data.curr_pt = 0  # number of data points actually acquired
+
+    first_pos = 1 if np else 0
+    first_det = np + 1
 
     # Defining a default scanPositioner Object for "Index" at for key=0:
     P0 = scanPositioner()
```

Comment:
It makes the code easier to read/review/... when the [Python naming conventions](https://pythonguides.com/python-naming-conventions/) (also [PEP8](https://peps.python.org/pep-0008/#naming-conventions)) are followed.  This `P0` looks like a global constant rather than a local variable.  Advise to change `P0` to `p0`.